src/components/interaction-designer/BuilderCanvas.vue

```diff
@@ -12,21 +12,54 @@
 </template>
 
 <script lang="ts">
-import { mapGetters } from 'vuex'
-import { Component, Vue } from 'vue-property-decorator'
+import { Component, Vue, Watch } from 'vue-property-decorator'
 import Block from '@/components/interaction-designer/Block.vue'
+import { namespace } from 'vuex-class'
+import {IBlock, IFlow} from '@floip/flow-runner'
+import { find, isEqual, cloneDeep } from 'lodash'
+import { IValidationStatus } from '@/store/validation'
 
-  @Component({
-    components: {
-      Block,
-    },
+const flowVuexNamespace = namespace('flow')
+const validationVuexNamespace = namespace('validation')
 
-    computed: {
-      ...mapGetters('flow', ['activeFlow']),
-    },
-
-  })
+@Component({
+  components: {
+    Block,
+  },
+})
 export default class BuilderCanvas extends Vue {
+  @Watch('activeFlow', { deep: true, immediate: true })
+  async onActiveFlowChanged(newFlow: IFlow) {
+    console.debug('active flow has changed, validating ...', newFlow)
+    await this.validate_flow({ flow: newFlow })
+  }
+
+  @Watch('blocksOnActiveFlowForWatcher', { deep: true, immediate: true })
+  async onBlocksInActiveFlowChanged(newBlocks: IBlock[], oldBlocks: IBlock[]) {
+    if (!!newBlocks && newBlocks.length > 0) {
```

Comment:
Due to typing, we shouldn't be able to get into the state where we need this first presence guard against a null `newBlocks`, should we? (eg. `!!newBlocks`)

Reply:
Low priority: Also, can we flip this to be a bail-early situation to lay patterns to avoid a Flock of Geese situation? 
```
if (newblocks.length === 0) {
  return
}
```


Flock of Geese situation:
```
>
    >
        >
            >
                >
                    >
                >
            >
        >
    >
>
```

![image](https://user-images.githubusercontent.com/58382/116593743-a8046b00-a8de-11eb-80eb-32982eb45295.png)


😹

Reply:
Nice one, thanks for the image 😻 , doing some changes here